`linkify()` should escape the `tok.content`

The current `linkify()` code at <https://github.com/jonschlinkert/markdown-toc/blob/master/index.js#L178>

```js
/**
 * Turn headings into anchors
 */

function linkify(tok, opts) {
  if (tok && tok.content) {
    var text = titleize(tok.content, opts);
    var slug = utils.slugify(tok.content, opts);
    if (tok.seen) {
      slug += '-' + tok.seen;
    }
    if (opts && typeof opts.linkify === 'function') {
      return opts.linkify(tok, text, slug, opts);
    }
    tok.content = utils.mdlink(text, '#' + slug);
  }
  return tok;
}
```

I think that the `slug` should be `querystring.escape(slug)`, otherwise the CJK characters may not correctly escaped for the generated URL.

After applied `querystring.escape(slug)`, one of the existing test failed:

```bash
  1) toc.insert should accept options:

      AssertionError: expected '---\nname: Test\n---\n# Test\n\n> This is a test!\n\n<!-- toc -->\n\n- [Quickstart](#quickstart)\n- [Options](#options)\n- [Usage examples](#usage%20examples)\n- [Contributing](#contributing)\n- [Author](#author)\n\n<!-- tocstop -->\n\n## Quickstart\nThis is the quickstart section.\n\n## Options\nThis is the options section.\n\n## Usage examples\nThis is the usage examples section.\n\n## Contributing\nThis is the Contributing section.\n\n## Author\nThis is the Author section.' to be '---\nname: Test\n---\n# Test\n\n> This is a test!\n\n<!-- toc -->\n\n- [Quickstart](#quickstart)\n- [Options](#options)\n- [Usage examples](#usage examples)\n- [Contributing](#contributing)\n- [Author](#author)\n\n<!-- tocstop -->\n\n## Quickstart\nThis is the quickstart section.\n\n## Options\nThis is the options section.\n\n## Usage examples\nThis is the usage examples section.\n\n## Contributing\nThis is the Contributing section.\n\n## Author\nThis is the Author section.'
      + expected - actual

       <!-- toc -->

       - [Quickstart](#quickstart)
       - [Options](#options)
      -- [Usage examples](#usage%20examples)
      +- [Usage examples](#usage examples)
       - [Contributing](#contributing)
       - [Author](#author)

       <!-- tocstop -->

      at Assertion.fail (node_modules/should/lib/assertion.js:92:17)
      at Assertion.Object.defineProperty.value (node_modules/should/lib/assertion.js:164:19)
      at Context.<anonymous> (test/test.js:377:17)
```

The test case is:

```js
  it('should accept options', function() {
    var str = read('test/fixtures/insert.md');
    strip(toc.insert(str, { slugify: function(text) {
      return text.toLowerCase();
    }})).should.equal(read('test/expected/insert-options.md'));
  });
```

It's replaced the `slugify()` with a custom function, which just convert to the lower case. And for `# Usage examples`, it will convert to `usage example`, which has a SPACE in it. 

I think it's a bug in the test case, should I fix the expected value to `usage%20example`? or add a special case in `linkify()`? Which one do you think should be the expected value?

--- a/index.js
+++ b/index.js
@@ -5,6 +5,7 @@
  */
 
 var utils = require('./lib/utils');
+var querystring = require('querystring');
 
 /**
  * expose `toc`
@@ -182,6 +183,7 @@ function linkify(tok, opts) {
     if (tok.seen) {
       slug += '-' + tok.seen;
     }
+    slug = querystring.escape(slug);
     if (opts && typeof opts.linkify === 'function') {
       return opts.linkify(tok, text, slug, opts);
     }

--- a/test/expected/insert-options.md
+++ b/test/expected/insert-options.md
@@ -9,7 +9,7 @@ name: Test
 
 - [Quickstart](#quickstart)
 - [Options](#options)
-- [Usage examples](#usage examples)
+- [Usage examples](#usage%20examples)
 - [Contributing](#contributing)
 - [Author](#author)
 

--- a/test/test.js
+++ b/test/test.js
@@ -82,6 +82,12 @@ describe('options: custom functions:', function() {
     actual.content.should.equal('- [Foo & Bar](#foo--bar)');
   });
 
+  it('should escape the CJK characters in linkify', function() {
+    toc('# 中文').content.should.equal('- [中文](#%E4%B8%AD%E6%96%87)');
+    toc('# かんじ').content.should.equal('- [かんじ](#%E3%81%8B%E3%82%93%E3%81%98)');
+    toc('# 한자').content.should.equal('- [한자](#%ED%95%9C%EC%9E%90)');
+  });
+
   it('should condense spaces in the heading text', function() {
     var actual = toc('# Some    Article');
     actual.content.should.equal('- [Some Article](#some----article)');

